### PR TITLE
YT streaming compatibility

### PR DESCRIPTION
**Pull request by SnazzyPine25**

### CODE REVIEW

File: lib/discordrb/bot.rb [hunk 1]
```diff
@@ -391,14 +391,15 @@ def parse_mention(mention)
...
+    # @param url [String, nil] The Stream URL to display as a stream. nil for no stream.
+    # @param type [Symbol, nil] The type of stream it is.
+    def update_status(idletime, game, url, type = nil)
       gateway_check
 
       @game = game
       @idletime = idletime
       @streamurl = url
-      type = url ? 1 : nil
+      type = [nil, :twitch, :youtube].indexOf(type) + 1
```
Comment (meew0) [inline lib/discordrb/bot.rb]: indexOf...? Are we JavaScript now? :P

Comment (SnazzyPine25) [inline lib/discordrb/bot.rb]: I just made a js bot and my brain just went @meew0

Comment (neko) [inline lib/discordrb/bot.rb]: how did you not pick it up in testing? you tested it right? ;3

Comment (SnazzyPine25) [inline lib/discordrb/bot.rb]: I got sidetracked @neko

File: lib/discordrb/bot.rb [hunk 2]
```diff
@@ -391,14 +391,15 @@ def parse_mention(mention)
...
+    # @param url [String, nil] The Stream URL to display as a stream. nil for no stream.
+    # @param type [Symbol, nil] The type of stream it is.
+    def update_status(idletime, game, url, type = nil)
       gateway_check
 
       @game = game
       @idletime = idletime
       @streamurl = url
-      type = url ? 1 : nil
+      type = [nil, :twitch, :youtube].index(type) + 1
```
Comment (meew0) [inline lib/discordrb/bot.rb]: This is off by one, twitch streaming is type 1, this returns type 2 for it.

File: lib/discordrb/events/presence.rb [hunk 3]
```diff
@@ -75,7 +79,7 @@ def initialize(data, bot)
       @game = data['game'] ? data['game']['name'] : nil
       @server = bot.server(data['guild_id'].to_i)
       @url = data['game'] ? data['game']['url'] : nil
-      @type = data['game'] ? data['game']['type'].to_i : nil
+      @type = data['game'] ? [:game, :game, :twitch, :youtube][data['game']['type'].to_i] : nil
```
Comment (meew0) [inline lib/discordrb/events/presence.rb]: Same for this.

**Comment by meew0** [comment]
Also, no matter what I try, I can't get this to actually show streaming on YouTube. Can you give me the data you tried it with?

**Comment by SnazzyPine25** [comment]
@meew0 Try getting a link of a ongoing stream, that might work

**Comment by meew0** [comment]
Done that, doesn't work. Neither do I ever receive a presence update with type 2, for that matter. What did you use for testing?

**Comment by meew0** [comment]
wat

**Pull request closed without merging**